python-package/xgboost/spark/model.py

```diff
@@ -1,28 +1,73 @@
-# type: ignore
 """Xgboost pyspark integration submodule for model API."""
 # pylint: disable=fixme, invalid-name, protected-access, too-few-public-methods
 import base64
+import logging
 import os
 import uuid
-
-from pyspark import SparkFiles, cloudpickle
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+)
+
+from pyspark import SparkContext, SparkFiles, cloudpickle
 from pyspark.ml.util import DefaultParamsReader, DefaultParamsWriter, MLReader, MLWriter
 from pyspark.sql import SparkSession
 
 from xgboost.core import Booster
 
 from .utils import get_class_name, get_logger
 
-
-def _get_or_create_tmp_dir():
+if TYPE_CHECKING:
```

Comment:
Why type checking only? These are hard dependencies of the spark module, you can import them freely. The spark module is not part of package import: https://github.com/dmlc/xgboost/blob/7375bd058b48714d689d8404f7fc76b094412886/python-package/xgboost/__init__.py#L18 . As a result, what you import here won't affect users that don't import the spark module.

Reply:
@trivialfis any thoughts on that?